custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -53,10 +58,20 @@ def do_predict(self, logger=None):
         with tempfile.NamedTemporaryFile(suffix=file_ext) as f:
             filestorage.save(f)
             f.flush()
-            out_data = self._predictor.predict(f.name)
+            if self._target_type == TargetType.TRANSFORM:
+                out_data = self._predictor.transform(f.name)
+            else:
+                out_data = self._predictor.predict(f.name)
 
         if self._target_type == TargetType.UNSTRUCTURED:
             response = out_data
+        elif self._target_type == TargetType.TRANSFORM:
+            if is_sparse(out_data):
+                mtx_payload = make_mtx_payload(out_data)
+                response = '{{"X.transformed":{mtx_payload}}}'.format(mtx_payload=mtx_payload)
```

Comment:
How are we going to know in DR if the payload is arrow or mtx?

Reply:
good point - how do you feel about changing the key name to have a sparse/dense indicator? something like X.transformed.sparse, X.transformed.dense?

Reply:
I'm not sure - is the response json? can we put something else in the json?

Reply:
yeah we could add another key to indicate sparse vs dense

Reply:
:+1: